### PR TITLE
Custom map names convert

### PR DESCRIPTION
**Pull request by gawquon**
Have CK3 counties with player overridden names transfer to their associated EU4 provinces in a sensible manner.
Also patch some typos and such in tradition_ideas.txt.

### CODE REVIEW

File: CK3toEU4/Source/EU4World/Province/EU4Province.cpp [hunk 1]
```diff
@@ -42,6 +43,16 @@ void EU4::Province::initializeFromCK3Title(const std::shared_ptr<CK3::Title>& or
...
 
+	// check for manual CK3 province name
+	if (srcProvince->isRenamed() && !srcProvince->isManualNameClaimed())
+	{
+		auto name = locDegrader.degradeString(srcProvince->getDisplayName());
+		details.customName = name;
+		// srcProvince seems to be an instance of "const shared_ptr<T>" and not "const shared_ptr<const T>, so we're using that"
+		srcProvince->setManualNameClaim();
+	}
+	mappers::LocDegraderMapper;
```
Comment (Zemurin) [inline CK3toEU4/Source/EU4World/Province/EU4Province.cpp:54]: What's going on here?

File: CK3toEU4/Source/EU4World/Province/EU4Province.cpp [hunk 2]
```diff
@@ -42,6 +43,16 @@ void EU4::Province::initializeFromCK3Title(const std::shared_ptr<CK3::Title>& or
 	details.owner = tagCountry.first;
 	details.controller = tagCountry.first;
 
+	// check for manual CK3 province name
+	if (srcProvince->isRenamed() && !srcProvince->isManualNameClaimed())
+	{
+		auto name = locDegrader.degradeString(srcProvince->getDisplayName());
```
Comment (Zemurin) [inline CK3toEU4/Source/EU4World/Province/EU4Province.cpp:49]: Locdegrader might not do anything on - say - cyrillic inputs. Yo should try renaming provinces into chinese and see what happens in eu4, just to be on the safe side.

**Review by Zemurin — CHANGES_REQUESTED** [review]
I'd also love to see some tests showing how renaming is done if possible.

File: CK3toEU4/Source/CK3World/Titles/Title.cpp [hunk 3]
```diff
@@ -80,6 +80,10 @@ void CK3::Title::registerKeys()
 	registerKeyword("holder", [this](const std::string& unused, std::istream& theStream) {
 		holder = std::pair(commonItems::singleLlong(theStream).getLlong(), nullptr);
 	});
+	registerKeyword("renamed", [this](std::istream& theStream) {
+		const auto& isRenamed = commonItems::getString(theStream);
+		renamed = isRenamed == "yes";
```
Comment (Idhrendur) [inline CK3toEU4/Source/CK3World/Titles/Title.cpp:85]:
```suggestion
		renamed = commonItems::getString(theStream)== "yes";
```

File: CK3toEU4/Source/EU4World/EU4World.cpp [hunk 4]
```diff
@@ -668,6 +668,18 @@ std::optional<std::pair<std::string, std::shared_ptr<CK3::Title>>> EU4::World::d
...
+	if (toReturn.second->isRenamed())
+	{
+		// ... if they belong to the same owner, Constantinople also has a custom name and they are in the same mapping
+		const auto& myDuchyCapital = toReturn.second->getDJLiege()->second->getCapital().second;
+		if (myDuchyCapital->isRenamed() && ck3Titles.contains(myDuchyCapital->getName()) &&
+			 myDuchyCapital->getDFLiege()->first == toReturn.second->getDFLiege()->first)
+		{
+			toReturn.second->overrideDisplayName(myDuchyCapital->getDisplayName());
+		}
+	}
```
Comment (Idhrendur) [inline CK3toEU4/Source/EU4World/EU4World.cpp:682]: In addition to the comment from @Zemurin, if you place this all in a small function you can use guard clauses for the checks and avoid deep nesting: https://refactoring.com/catalog/replaceNestedConditionalWithGuardClauses.html

**Comment by Wielislaw** [comment]
If it actually works, then it could and should be implemented to the rest of the converters.

File: CK3toEU4/Source/EU4World/Province/EU4Province.cpp [hunk 5]
```diff
@@ -42,6 +48,9 @@ void EU4::Province::initializeFromCK3Title(const std::shared_ptr<CK3::Title>& or
 	details.owner = tagCountry.first;
 	details.controller = tagCountry.first;
 
+	// check for manual CK3 province name
```
Comment (Zemurin) [inline CK3toEU4/Source/EU4World/Province/EU4Province.cpp:51]: Was there a check supposed to be here?

File: CK3toEU4/Source/CK3World/Titles/Title.cpp [hunk 6]
```diff
@@ -274,6 +277,40 @@ std::map<std::string, std::shared_ptr<CK3::Title>> CK3::Title::coalesceDJCountie
 	return toReturn;
 }
 
+void CK3::Title::pickDisplayName(const std::map<std::string, std::shared_ptr<Title>>& mappings)
```
Comment (Zemurin) [inline CK3toEU4/Source/CK3World/Titles/Title.cpp:280]: Please don't use generic "mappings" but something more appropriate like "availableTitles" or such.

**Review by Zemurin — CHANGES_REQUESTED** [review]
Just cosmetics.

**Review by Zemurin — APPROVED** [review]

**Pull request merged**